src/DynamicData.Tests/List/MergeManyCacheChangeSetsFixture.cs

```diff
@@ -0,0 +1,737 @@
+        highPriceResults.Summary.Overall.Refreshes.Should().Be(0);
+        highPriceResults.Data.Items.Select(cp => cp.MarketId).ForEach(guid => guid.Should().Be(marketOriginal.Id));
+    }
+
+    [Fact]
+    public void EqualityComparerHidesUpdatesWithoutChanges()
+    {
+        // having
+        var market = new Market(0);
+        using var results = _marketList.Connect().MergeManyChangeSets(m => m.LatestPrices, MarketPrice.EqualityComparer).AsAggregator();
+        market.SetPrices(0, PricesPerMarket, LowestPrice);
+        _marketList.Add(market);
+
+        // when
+        market.UpdateAllPrices(LowestPrice);
+
+        // then
+        _marketListResults.Data.Count.Should().Be(1);
+        results.Data.Count.Should().Be(PricesPerMarket);
+        results.Messages.Count.Should().Be(1);
+        results.Summary.Overall.Adds.Should().Be(PricesPerMarket);
+        results.Summary.Overall.Removes.Should().Be(0);
+        results.Summary.Overall.Updates.Should().Be(0);
+        results.Summary.Overall.Refreshes.Should().Be(0);
+    }
+
+    [Fact]
+    public void EveryItemVisibleWhenSequenceCompletes()
+    {
+        // having
+        _marketList.AddRange(Enumerable.Range(0, MarketCount).Select(n => new FixedMarket(GetRandomPrice, n * ItemIdStride, (n * ItemIdStride) + PricesPerMarket)));
+
+        // when
+        using var results = _marketList.Connect().MergeManyChangeSets(m => m.LatestPrices).AsAggregator();
+        DisposeMarkets();
+
+        // then
+        results.Data.Count.Should().Be(PricesPerMarket * MarketCount);
+        results.Summary.Overall.Adds.Should().Be(PricesPerMarket * MarketCount);
+        results.Summary.Overall.Removes.Should().Be(0);
+        results.Summary.Overall.Updates.Should().Be(0);
+        results.Summary.Overall.Refreshes.Should().Be(0);
+    }
+
+    [Theory]
+    [InlineData(false, false)]
+    [InlineData(false, true)]
+    [InlineData(true, false)]
+    [InlineData(true, true)]
+    public void MergedObservableCompletesOnlyWhenSourceAndAllChildrenComplete(bool completeSource, bool completeChildren)
+    {
+        // having
+        _marketList.AddRange(Enumerable.Range(0, MarketCount).Select(n => new FixedMarket(GetRandomPrice, n * ItemIdStride, (n * ItemIdStride) + PricesPerMarket, completable: completeChildren)));
+        var hasSourceSequenceCompleted = false;
+        var hasMergedSequenceCompleted = false;
+
+        using var cleanup = _marketList.Connect().Do(_ => { }, () => hasSourceSequenceCompleted = true)
+                            .MergeManyChangeSets(m => m.LatestPrices).Subscribe(_ => { }, () => hasMergedSequenceCompleted = true);
+
+        // when
+        if (completeSource)
+        {
+            DisposeMarkets();
+        }
+
+        // then
+        hasSourceSequenceCompleted.Should().Be(completeSource);
+        hasMergedSequenceCompleted.Should().Be(completeSource && completeChildren);
+    }
+
+    [Fact]
+    public void MergedObservableWillFailIfSourceFails()
```

Comment:
Should we also test that errors from child streams propagate out?

Reply:
I decidedly did not want errors from child streams to tear down the whole thing.  I could be convinced that is the wrong behavior, but my thinking was that an error on the child stream shouldn't disrupt all of the other child streams or the main stream.